extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -447,23 +535,92 @@ export class AmpAutocomplete extends AMP.BaseElement {
    * @private
    */
   inputHandler_() {
+    let inlinePromise = Promise.resolve();
+
     // If the input is the first entry in the field.
     const firstEntry =
       this.userInput_.length === 0 && this.inputElement_.value.length === 1;
     this.userInput_ = this.inputElement_.value;
-    return this.mutateElement(() => {
-      this.filterDataAndRenderResults_(this.sourceData_, this.userInput_);
-      this.toggleResults_(true);
-      if (this.suggestFirst_) {
-        if (!this.detectBackspace_ || firstEntry) {
-          this.updateActiveItem_(1);
-        }
-        // Reset flag.
-        this.detectBackspace_ = false;
+
+    if (this.inline_) {
+      const match = this.getClosestPriorMatch_(this.trigger_);
+      this.triggered_ = !!match;
+
+      if (!this.triggered_) {
+        return this.mutateElement(() => {
+          this.clearAllItems_();
+        });
+      }
+
+      this.match_ = match;
+      this.userInput_ = this.match_[0].slice(this.trigger_.length);
+
+      if (!this.triggered_ && !this.menuTriggered_) {
+        return Promise.resolve();
       }
+      let hadResultsBeforeThisInput = false;
+      if (this.sourceData_ && this.userInput_ != '') {
+        hadResultsBeforeThisInput = true;
+      }
+
+      inlinePromise = this.getRemoteData_().then(sourceData => {
+        this.sourceData_ = sourceData || [];
+        this.menuTriggered_ = this.sourceData_.length;
+        if (!this.menuTriggered_ && hadResultsBeforeThisInput) {
+          this.triggered_ = false;
+        }
+      });
+    }
+
+    return inlinePromise.then(() => {
+      return this.mutateElement(() => {
+        this.filterDataAndRenderResults_(
+          this.sourceData_,
+          this.userInput_
+        ).then(() => {
+          this.toggleResults_(true);
+          if (this.suggestFirst_) {
+            if (!this.detectBackspace_ || firstEntry) {
+              this.updateActiveItem_(1);
+            }
+            // Reset flag.
+            this.detectBackspace_ = false;
+          }
+        });
+      });
     });
   }
 
+  /**
+   * Finds the closest string in the user input prior to the cursor
+   * to display suggestions.
+   * @param {string} trigger
+   * @return {?RegExpResult}
+   * @private
+   */
+  getClosestPriorMatch_(trigger) {
+    const delimiter = `\\${trigger}`;
```

Comment:
So what happens if I include special regex chars in my trigger string?

Reply:
This escapes all chars which works for regex chars and breaks everything else. I updated it to escape on the regex chars only. This fixes it for the current case (providing a string) but introduces a new level of complexity for the future case (providing a regex expression). Hmm..

Reply:
Should we maybe explicitly declare a regex then? Either via an attribute flag or some sort of DSL on the current attribute (would probably prefer the former).